torch/testing/_internal/common_device_type.py

```diff
@@ -271,6 +274,11 @@ def instantiated_test(self, name=name, test=test_fn, dtype=dtype, op=op):
                     self.precision = self._get_precision_override(test_fn, dtype)
                     args = (arg for arg in (device_arg, dtype, op) if arg is not None)
                     result = test_fn(self, *args)
+                except RuntimeError as rte:
+                    if 'CUDA error: device-side assert triggered' in rte.__repr__():
```

Comment:
@ngimel: there are other CUDA error strings, too, like the illegal memory access string

@ngimel: also, math libaries like cuDNN may wrap context-destroying errors and throw errors with different strings

Reply:
I think we should redesign this check a bit. Instead of trying to check for particular error strings, when an errors occurs let's check here if the test suite should continue or not with a helper function.

```
except RuntimeError as rte:
  self._stop_test_suite = _should_stop_test_suite(rte)
```

In the base, the implementation of this function is trivial:

```
def _should_stop_test_suite(self, rte):
  return False
```

CUDATestBase, however, can override this function like this:

```
def _should_stop_test_suite(self, rte):
  try:
    torch.cuda.synchronize()
  except RuntimeError as e:
    return True
  return False
```

This may be an imperfect approach but @ngimel and I think it's a good start and likely to catch most cases. This also only synchronizes if there actually is an error. We wouldn't want to synchronize generally since then running tests in parallel would not work as expected.

Reply:
good idea. one quick follow up - are all CUDA device-side errors wrapped as rte when passed back to python?